src/solvers/refinement/string_constraint_generator_indexof.cpp

```diff
@@ -211,10 +385,10 @@ exprt string_constraint_generatort::add_axioms_for_last_index_of(
 
   // We add axioms:
   // a1 : -1 <= i <= from_index
-  // a2 : (i=-1 <=> !contains)
-  // a3 : (contains => i <= from_index &&s[i]=c)
-  // a4 : forall n. i+1 <= n < from_index +1 &&contains => s[n]!=c
-  // a5 : forall m. 0 <= m < from_index +1 &&!contains => s[m]!=c
+  // a2 : i = -1 <=> !contains
+  // a3 : contains ==> (i <= from_index && s[i] = c)
+  // a4 : forall n:[i+1, from_index+1[ && contains ==> s[n] != c
+  // a5 : forall m:[0, from_index+1[ && !contains ==> s[m] != c
```

Comment:
`String.lastIndexOf()` still has a problem. CBMC is non-terminating on the following example:
```
      String s = "Abc";
      String bc = "bc";
      int i = s.lastIndexOf(bc);
      assert(i == 1);
```
`String.indexOf()` now seems to work nicely however.